stresstests/blobfusestress.cpp

```diff
@@ -0,0 +1,619 @@
+    std::cout << "Running small file stress test." << std::endl;
+    print_test_initial_stats(total_dir_count, file_per_dir_count, file_size_base, additional_size_jitter);
+
+    for (int i = 0; i < total_dir_count; i++)
+    {
+        std::string dir = source_dir + "/" + std::to_string(i);
+        int mkdirret = mkdir(dir.c_str(), 0777);
+        if (mkdirret < 0)
+        {
+            std::stringstream error;
+            error << "Failed to make directory.  errno = " << errno << ", directory = " << dir;
+            throw std::runtime_error(error.str());
+        }
+
+        std::string file = dir + "/file";
+
+        uint_fast32_t r_local_seed = r();
+        pool.add_task([=] () {
+            std::minstd_rand r_local(r_local_seed);
+            for (int j = 0; j < file_per_dir_count; j++)
+            {
+                std::stringstream file_name_stream;
+                file_name_stream << file << std::setfill('0') << std::setw(8) << j;
+                size_t file_size = file_size_base + (r_local() % additional_size_jitter);
+                uint_fast32_t current = r_local();
+                std::ofstream file_stream(file_name_stream.str(), std::ios::binary);
+                for (size_t i = 0; i < file_size; i += 4 /* sizeof uint_fast32_t */)
+                {
+                    file_stream.write(reinterpret_cast<char*>(&current), 4);
+                    current++;
+                }
+            }
+        });
+    }
+
+    pool.drain();
+    return std::make_pair(total_size, total_dir_count * file_per_dir_count);
+}
+
+int main(int argc, char *argv[])
+{
+    std::vector<std::function<std::pair<size_t, size_t>(std::string, thread_pool&)>> populate_fns
+    {
+        populate_small,
+        populate_large,
+    };
+
+    std::cout << populate_fns.size() << " tests to run in total." << std::endl << std::endl;
+    for (int i = 0; i < populate_fns.size(); i++)
+    {
+        std::cout << std::endl << "Starting test " << i << "." << std::endl;
+        std::function<std::pair<size_t, size_t>(std::string, thread_pool&)> populate_func = populate_fns[i];
+
+        std::time_t start = std::chrono::high_resolution_clock::to_time_t(std::chrono::high_resolution_clock::now());
+        std::cout << "Start time = " << std::ctime(&start) << std::endl;
+        std::string source_dir("/mnt/resource/tests/src");  // Source directory for test data.  This is an SSD on my machine.  Should not be a blobfuse directory.  All contents will be wiped.
+        std::string dest_dir_1("/home/asorrin/code/azure-storage-fuse/build/mountdir/stress");  // blobfuse directory to copy to.  All contents will be wiped.
+        std::string dest_dir_2("/mnt/resource/tests/dst");  // Local destination directory.  This is an SSD on my machine.  Should not be a blobfuse directory.  All contents will be wiped.
```

Comment:
Pull these strings to the top and make them not-my-machine specific.  (Or better, put in a config file, if it's easy)